docs/source/design/ui_design.rst

```diff
@@ -0,0 +1,328 @@
+programming is preferred over object oriented style as described in their `code
+standard <https://pyscf.org/code-rule.html>`_.
+
+Below you can find how to run an SCF calculation for a hydrogen molecule using
+PySCF.
+
+.. code-block:: python
+
+    from pyscf import gto, scf
+    mol = gto.M(atom='O 0 0 0; H 0 1 0; H 0 0 1', basis='sto-3g')
+    rhf = scf.RHF(mol)
+    energy = rhf.kernel()
+
+Here, mol is the object (type ``pyscf.gto.mole.Mole``) representing the molecule
+and integrals with the given basis set and rhf is the object (type
+``pyscf.scf.hf.RHF``) that holds method specific information. The energy is
+computed using the ``kernel()`` function.
+
+PSI4
+====
+PSI4 is a C++/Python (70%/30%) open-source quantum chemistry package distributed
+under the LGPL3 license. PSI4 provides two different types of UI referred to as
+Psithon and PsiAPI modes. In the Psithon mode, the user writes an input file in
+a domain specific language similar to Python. In the PsiAPI mode, the user can
+write a pure Python script that interacts with PSI4 as a Python module. Since
+the latter is more relevant to our design, we show below how to run an SCF
+calculation for a hydrogen molecule using the PsiAPI mode.
+
+.. code-block:: python
+
+    import psi4
+    mol = psi4.geometry('H 0. 0. 0. \n H 0. 0. 1.')
+    energy= psi4.energy('scf/sto-3g')    
+
+Here, mol is the molecule object, which is created using the ``psi4.geometry()``
+function and the energy is computed using the ``psi4.energy()`` function, which
+takes the method name as the required argument. Note that the user do not need
+to pass the molecule object explicitly to the subsequent energy calculation. By
+default energy is computed for the last molecule defined with the
+``psi4.geometry()`` function. Additional options can be set with the
+``psi4.set_options()`` function, which takes a Python dictionary as the required
+argument. 
+
+QCEngine
+========
+QCEngine is a general purpose quantum chemistry program interface. It is a
+Python library that provides a common API for quantum chemistry programs.
+QCEngine is an open-source package distributed under the Apache License 2.0.
+
+Below you can find how to run an SCF calculation for a hydrogen molecule using
+QCEngine.
+
+.. code-block:: python
+
+    import qcengine as qcng
+    import qcelemental as qcel
+    mol = qcel.models.Molecule.from_data('H 0. 0. 0. \n H 0. 0. 1.')
+    out = qcng.compute({"molecule": mol, "driver": "energy", "model": {"method":
+    "SCF", "basis": "sto-3g"}}, "NWChemEx")
+    energy = out.return_result
+
+Here, mol is the molecule object (type
+``qcelemental.models.molecule.Molecule``), which is created using the
+``qcel.models.Molecule.from_data()`` function from QCElemental package. The SCF
+energy is computed using the ``qcng.compute()`` function, from QCEngine
+package. Note that, the input for the ``qcng.compute`` function is a Python
+dictionary with a schema defined by QCElemental.
+
+NWChemEx
```

Comment:
I'd rename this to `Current NWChemEx UI Design` or something similar to make it more clear that this is the real "meat and potatoes" of the page.

Reply:
Fixed with https://github.com/NWChemEx-Project/.github/commit/c28a7cb41f1d7d9cb6d97078ae526af84a8eb86f.